src/main/java/practice/StreamPractice.java

```diff
@@ -23,7 +35,10 @@ public int findMinEvenNumber(List<String> numbers) {
      * But before that subtract 1 from each element on an odd position (having the odd index).
      */
     public Double getOddNumsAverage(List<Integer> numbers) {
-        return 0D;
+        return IntStream.range(0, numbers.size())
+                .map(n -> n % 2 == 0 ? numbers.get(n) : numbers.get(n) - 1)
+                .filter(n -> n % 2 != 0)
+                .average().orElseThrow(NoSuchElementException::new);
```

Comment:
first of all, I think, you can move last expression (.orElseThrow) to the next line.
And secondly I think that you can simplify it by replacing .orElseThrow to just .getAsDouble(), Because .getAsDouble() will throw NoSuchElementException in case empty number
```suggestion
                .average()
                .getAsDouble();
```